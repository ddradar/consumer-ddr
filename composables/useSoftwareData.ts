import type { ParsedContent } from '@nuxt/content/dist/runtime/types'
import type { Ref } from 'vue'

import type seriesList from '~~/content/series/.seriesList.json'

export const regionList = ['JP', 'US', 'EU', 'None'] as const

export const platforms = [
  'Play Station',
  'Play Station 2',
  'Dreamcast',
  'GAME BOY COLOR',
  'GAME CUBE',
  'Wii',
  'Xbox 360',
  'Windows',
] as const

export interface Software {
  slug: keyof typeof seriesList
  title: string
  platform: (typeof platforms)[number]
  region: (typeof regionList)[number]
  launched: string
  difficulties: { [key in number]: { name: string; class: string } }
}

export default async function (slug: string) {
  const { data } = await useAsyncData(`/software/${slug}`, () =>
    queryContent<Software & ParsedContent>()
      .where({ _type: 'markdown', slug })
      .findOne()
  )

  return { software: data as Ref<Software & ParsedContent> }
}
