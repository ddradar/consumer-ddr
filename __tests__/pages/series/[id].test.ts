import { mockNuxtImport } from 'nuxt-vitest/utils'
import { describe, expect, test, vi } from 'vitest'
import { ref } from 'vue'

import useSoftwareData from '~~/composables/useSoftwareData'
import SeriesDetail from '~~/pages/series/[id].vue'

import { mockSoftware, mountAsync, plugins } from '../../test-utils'

vi.mock('~~/composables/useSoftwareData')
mockNuxtImport('useRoute', () => () => ({ params: { id: mockSoftware.slug } }))

describe('pages/series/[id].vue', () => {
  const stubs = { ContentRenderer: true, Head: true, Title: true }
  const global = { plugins, stubs }

  describe('snapshot test', () => {
    test('renders correctly', async () => {
      // Arrange
      vi.mocked(useSoftwareData).mockResolvedValue({
        software: ref(mockSoftware),
      })

      // Act
      const wrapper = await mountAsync(SeriesDetail, { global })

      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
