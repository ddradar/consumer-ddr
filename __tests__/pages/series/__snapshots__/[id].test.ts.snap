// Vitest Snapshot v1

exports[`pages/series/[id].vue > snapshot test > renders correctly 1`] = `
<section
  class="section"
>
  <head-stub />
  <content-renderer-stub
    class="content"
    value="[object Object]"
  />
</section>
`;
